src/components/Form.jsx

```diff
@@ -35,109 +36,92 @@ export const Checkbox = ({ id, label, checked, disabled, onChange }) => (
     </span>
 );
 
-export class Slider extends React.Component {
-    onChange = val => {
-        this.props.onChange(val.currentTarget.value);
-    };
-
-    shouldComponentUpdate(nextProps) {
-        return this.props.value !== nextProps.value;
-    }
-
-    render() {
-        const { min = 1, max = 10, step = 1, value } = this.props;
+export const Slider = ({ min = 1, max = 10, step = 1, value, onChange }) => {
+    const getActiveWidth = useMemo(() => {
         const barWidth = 238; // css hardcode
-        let activeWidth = Math.round((barWidth * (value - min)) / (max - min));
-        activeWidth = activeWidth < 0 ? 0 : activeWidth;
+        const activeWidth = Math.round((barWidth * (value - min)) / (max - min));
+        return activeWidth < 0 ? 0 : activeWidth;
+    }, [value, min, max]);
 
-        return (
-            <div className='sc-slider'>
-                <div className='sc-slider-range'>
-                    <div className='sc-slider-bar' />
-                    <div className='sc-slider-active-bar' style={{ width: activeWidth }} />
-                    <input type='range' min={min} max={max} step={step} onChange={this.onChange} value={value} />
-                </div>
-                <div className='value'>{value}</div>
+    const handleChange = useCallback(el => onChange(el.currentTarget.value), [onChange]);
+
+    return (
+        <div className='sc-slider'>
+            <div className='sc-slider-range'>
+                <div className='sc-slider-bar' />
+                <div className='sc-slider-active-bar' style={{ width: getActiveWidth }} />
```

Comment:
should we call this function `getActiveWidth`?

Reply:
@balakrishna-binary it's `useMemo`

Reply:
maybe we should rename it to `activeWidth`?
`getActiveWidth` sound like a function

Reply:
@balakrishna-binary agreed. just updated it.